Code Developed Between GA and Bedford

strimper13 opened the pull request:
NOM NOM NOM AM HUNGER

strimper13 committed:
Changed Slow Turn Command in Drive

Fixed slow turn speeds to match at .5 & -.5 respectively

Quinn-5 committed:
Implement LEDs on beak open and close, reformat and organize inputs

Quinn-5 committed:
Stop uploading unformatted code

Reformat any files that needed it

Rohan-Sreenivasan committed:
Led stuff is fixed

All the led comments is fine

Rohan-Sreenivasan committed:
good now

i lied its good now

Rohan-Sreenivasan committed:
List

Leds use a list now so its easier to call a color

Rohan-Sreenivasan committed:
Leds Blink

There is now a command to make the leds blink and its great

Rohan-Sreenivasan committed:
Led stuff

Stuff is better, we can pass over every member in the array and turn on the banks

andrewmartin5 committed:
added ramp up to the lift motor

hopefully its less jolty now

andrewmartin5 committed:
Made pivot point disappear and replaced it with a piston

andrewmartin5 committed:
Button assigning and added intake and eject command

andrewmartin5 committed:
Command to blink Vision lights

strimper13 committed:
Merge branch 'postGA'

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -161,9 +163,21 @@ public Command goToCargoShipCargo() {
     }
 
+    public CommandChain LEDOpenBeak() {
+        CommandChain retValue = new CommandChain("Close Beak and Turn n Green LEDs");
```
Probably could make this shorter - also "turn n green" is missing a letter

strimper13 replied:
no problemo mi amigo. yo estaba llorando y haciendo la coda.

msoucy replied:
Mi ne parolas la hispanan. Ĉu vi parolas Esperanton?

msoucy replied:
Also, apparently "code" in Spanish is "código". So -5 points for you. 😞

strimper13 replied:
yo no se mi amigo

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -1,5 +1,7 @@
 package frc.robot;
 
+import java.util.Arrays;
```
Is this import needed? (Press `alt-shift-o` to automatically clean up imports)

strimper13 replied:
Handy tip, thank you and done

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -15,9 +17,9 @@
 import frc.robot.subsystems.Drive;
 import frc.robot.subsystems.Leds;
 import frc.robot.subsystems.LiftSubsystem;
+import frc.robot.subsystems.LiftSubsystem.Heights;
```
If you import `Heights` like this, you don't need line 126 to say `LiftSubsystem.Heights.kLiftFlipHeight`, you can just say `Heights.kLiftFlipHeight`

strimper13 replied:
done-zo

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -161,9 +163,21 @@ public Command goToCargoShipCargo() {
     public void assignButtons() {
-        xBoxCoPilot.getButton(ButtonXboxController.XBoxButton.BUMPER_LEFT).whenPressed(manipulator.openBeak());
-        xBoxCoPilot.getButton(ButtonXboxController.XBoxButton.BUMPER_RIGHT.get()).whenPressed(manipulator.closeBeak());
+        xBoxCoPilot.getButton(ButtonXboxController.XBoxButton.BUMPER_LEFT).whenPressed(LEDOpenBeak());
```
Add a `import com.chopshop166.chopshoplib.controls.ButtonXboxController.XBoxButton;` at the top and you should be able to make these lines shorter.

strimper13 replied:
gotcha

msoucy commented on src/main/java/frc/robot/maps/CurrentRobot.java:
```diff
@@ -88,6 +84,11 @@ public DoubleSolenoid getbeaksPiston() {
+
+            @Override
+            public DoubleSolenoid getArmsPiston() {
+                return null;
```
I feel like this one shouldn't be `null`...

strimper13 replied:
Bam Done

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -71,14 +78,17 @@ protected boolean isFinished() {
-                ldrive_can.SetColor(2, Color.GREEN, (float) 0.8);
+                for (Integer currentBank : banks) {
+                    ldrive_can.SetColor(currentBank, Color.GREEN, 1.0);
+
```
These blank lines make it look weird

strimper13 replied:
Done, saved 60 lines of soldiers from their inactive duty

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -143,25 +142,71 @@ protected boolean isFinished() {
+
+            protected void initialize() {
+                counter = 0;
+                // for (Integer currentBank : banks) {
```
Don't have commented-out code

strimper13 replied:
got it

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -143,25 +142,71 @@ protected boolean isFinished() {
 
-    public Command killAllLights() {
+    public Command blinkLights(Color color, int frequency, Integer... banks) {
         // The command is named "Sample Command" and requires this subsystem.
```
Fix comment

strimper13 replied:
killed the sample comments?

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -229,4 +269,54 @@ protected void interrupted() {
     }
+
+    public Command turnOnVisionLights() {
+        return new InstantCommand("turn on the color known as vision", this, () -> {
```
Could probably make these command names shorter

strimper13 replied:
Yes, Darn it rohan

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,57 @@ public Command rollerStop() {
-                pivotPointsMotor.set(.2);
-            }
+    public Command gamepieceCheck() {
+        // This command will pick up a ball
```
Will it, though? Or will it just wait until a game piece is in position?

strimper13 replied:
Rohan sits on a seat of lies

strimper13 replied:
or andrew, just salty about fixing rohans space

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,57 @@ public Command rollerStop() {
-        // This command will pick up a ball
-        return new Command("Check for Gamepiece", this) {
+    public TimedCommand Eject() {
+        return new TimedCommand("Eject Ball", 1) {
```
`return new TimedCommand("Eject Ball", 1, this)` - you need to add `this` otherwise it'll cause scheduler problems

strimper13 replied:
Thank you, done

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,57 @@ public Command rollerStop() {
             @Override
             protected boolean isFinished() {
-                return gamepieceLimitSwitch.get();
+                return false;
```
Don't override `isFinished` for a `TimedCommand`, that defeats the purpose

strimper13 replied:
killed isFinished

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,57 @@ public Command rollerStop() {
             @Override
             protected boolean isFinished() {
-                return intakePositionLimitSwitch.get();
+                return false;
```
- When does this command get run?
- When does it finish?

strimper13 replied:
done

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -229,4 +269,54 @@ protected void interrupted() {
+            @Override
+            protected void initialize() {
+                counter = 0;
+                ldrive_can.SetColor(3, Color.GREEN, 1.0);
```
Would you kindly make this 3 a constant up above so we know which bank it represents?

strimper13 replied:
done, named it visionBank

strimper13 replied:
done

strimper13 committed:
Code for Bedford day One

FIxed all comments on pull request

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -54,7 +49,6 @@ public Command setTeamColor(Integer... banks) {
-                // Called just before this Command runs the first time
                 Color color;
 
                 if (isBlueTeam()) {
```
Why does it determine color if it sets it to `Color.OFF` in the loop?

strimper13 replied:
It does a good?

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -72,104 +66,85 @@ protected void initialize() {
             protected void execute() {
-                // Called repeatedly when this Command is scheduled to run
                 for (Integer currentBank : banks) {
                     ldrive_can.SetColor(currentBank, Color.BLUE, 1.0);
```
This says it sets the color to `Color.BLUE`...

strimper13 replied:
Yes?

strimper13 committed:
Shuffle board match details - Rohan

Match details can be recorded and we can see the type, time and event name

strimper13 committed:
Backwards drive and kill switch - Rohan

I made a command to drive back and it has a toggle button. We also got a button to kill the current command

strimper13 committed:
PID Lift - Rohan

The lift pid was kinda tuned

strimper13 committed:
Olivia writes comments

strimper13 committed:
Changed manipulator returns to double solenoid

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -102,6 +102,24 @@ protected boolean isFinished() {
         };
     }
 
+    public Command driveBackWards() {
```
`BackWards` is one word and shouldn't captialize the W

strimper13 replied:
bAcKwArDs

msoucy commented on src/main/java/frc/robot/subsystems/Leds.java:
```diff
@@ -1,31 +1,31 @@
     Color fuschia = new Color(255, 0, 255);
+    private final static int leftBank = 1;
+    private final static int rightBank = 2;
+    private final static int visionBank = 3;
```
These are constants, and should probably be named `LEFT_BANK` etc. in the correct nomenclature of constants.

strimper13 replied:
I_FINISHED_THIS

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,52 @@ public Command rollerStop() {
     }
+    // #endregion
+
+    public Command Intake() {
```
Lowercase the `I`

strimper13 replied:
done

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -117,53 +131,52 @@ public Command rollerStop() {
-    public Command gamepieceCheck() {
-        // This command will pick up a ball
-        return new Command("Check for Gamepiece", this) {
+    public TimedCommand Eject() {
```
Lowercase the `E`

strimper13 replied:
e-verytime

msoucy commented on src/Vision/testpi.py:
```diff
@@ -376,3 +376,27 @@ def normalizeImage(pairMidpoint):
             table.putBoolean("Vision Found", False)
 
         outputmodeeam.putFrame(frame)
+
```
![image](https://user-images.githubusercontent.com/1126295/54726727-c23dbd80-4b4a-11e9-93fb-745d61787499.png)

strimper13 replied:
Twas' Olivia

msoucy reviewed (changes requested):
So close...

strimper13 committed:
Fixed comments

closes #21 :)

strimper13 committed:
Fixed BackWards again
